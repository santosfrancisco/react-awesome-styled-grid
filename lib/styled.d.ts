/* eslint-disable @typescript-eslint/consistent-type-definitions */
// import original module declarations
import "styled-components";
import { Config } from "./types";
import { CUSTOM_CONF } from "../lib/config";

// and extend them!
declare module "styled-components" {
  export interface DefaultTheme {
    [CUSTOM_CONF]: Partial<Config>;
  }
}
