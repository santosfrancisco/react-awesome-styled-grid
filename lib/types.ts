import { CUSTOM_CONF } from "./config";
import * as React from "react";
import { DefaultTheme, css } from "styled-components";
export type GridBreakpoints = "xs" | "sm" | "md" | "lg" | "xl";

export type AlignOrJustifyOptions = {
  xs?: string;
  sm?: string;
  md?: string;
  lg?: string;
  xl?: string;
};

export type OffsetOptions = {
  xs?: number;
  sm?: number;
  md?: number;
  lg?: number;
  xl?: number;
};

export type OrderOptions = {
  xs?: number;
  sm?: number;
  md?: number;
  lg?: number;
  xl?: number;
};

export type Grid = {
  children: React.ReactNode;
  style?: React.CSSProperties;
  debug?: boolean;
};

export type GridContainerProps = Grid & {
  fluid?: boolean;
};

export type GridRowProps = Grid & {
  reverse?: boolean | GridBreakpoints[];
  align?: string | object;
  justify?: string | object;
};

export type GridColProps = Grid & {
  xs?: number | string;
  sm?: number | string;
  md?: number | string;
  lg?: number | string;
  xl?: number | string;
  offset?: number | OffsetOptions;
  order?: number | OrderOptions;
  reverse?: boolean | GridBreakpoints[];
  align?: string | AlignOrJustifyOptions;
  justify?: string | AlignOrJustifyOptions;
  noGutter?: boolean;
};

export type VisibleProps = {
  xs?: boolean;
  sm?: boolean;
  md?: boolean;
  lg?: boolean;
  xl?: boolean;
  children: React.ReactNode;
};

export type HiddenProps = {
  xs?: boolean;
  sm?: boolean;
  md?: boolean;
  lg?: boolean;
  xl?: boolean;
  children: React.ReactNode;
};

export type ScreenClassProps = {
  render: (screen: GridBreakpoints) => React.ReactNode;
};

export type Media = Record<GridBreakpoints, typeof css>;

export type Config = {
  media: Media;
  mediaQuery: string;
  columns: Record<GridBreakpoints, number>;
  gutterWidth: Record<GridBreakpoints, number>;
  paddingWidth: Record<GridBreakpoints, number>;
  container: Record<GridBreakpoints, number | string>;
  breakpoints: Record<GridBreakpoints, number>;
};

export type AwesomeTheme = { [CUSTOM_CONF]: Partial<Config> };

export type ConfigProps = { theme?: DefaultTheme };

export type Util = {
  getScreenClass: (props?: ConfigProps) => GridBreakpoints;
  getViewPort: () => number | null;
};
