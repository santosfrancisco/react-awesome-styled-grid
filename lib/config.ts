import { Config, ConfigProps, GridBreakpoints, Media } from "@site/types";
import { css, CSSObject } from "styled-components";

export const CUSTOM_CONF = "awesomegrid";
export const DIMENSIONS: GridBreakpoints[] = ["xs", "sm", "md", "lg", "xl"];
export const BASE_CONF: Config = {
  media: null,
  mediaQuery: "only screen",
  columns: {
    xs: 4,
    sm: 8,
    md: 8,
    lg: 12,
    xl: 12
  },
  gutterWidth: {
    xs: 1,
    sm: 1,
    md: 1.5,
    lg: 1.5,
    xl: 1.5
  },
  paddingWidth: {
    xs: 1,
    sm: 1,
    md: 1.5,
    lg: 1.5,
    xl: 1.5
  },
  container: {
    xs: "full", // 'full' = max-width: 100%
    sm: "full", // 'full' = max-width: 100%
    md: "full", // 'full' = max-width: 100%
    lg: 90, // max-width: 1440px
    xl: 90 // max-width: 1440px
  },
  breakpoints: {
    xs: 1,
    sm: 48, // 768px
    md: 64, // 1024px
    lg: 90, // 1440px
    xl: 120 // 1920px
  }
};

const configs = [];
const hasCustomConf = (props: ConfigProps) => {
  if (!props?.theme) return {};
  return JSON.stringify(props.theme[CUSTOM_CONF] || {});
};
const resolveConfig = (props: ConfigProps) => {
  let themeConf = {};
  if (props?.theme) {
    themeConf = props.theme[CUSTOM_CONF] || {};
  }

  const conf: Config = {
    ...BASE_CONF,
    ...themeConf
  };

  conf.media = Object.keys(conf.breakpoints).reduce((media, breakpoint) => {
    const breakpointWidth = conf.breakpoints[breakpoint];
    media[breakpoint] = makeMedia(
      [
        conf.mediaQuery,
        breakpointWidth >= 0 && `(min-width: ${breakpointWidth}rem)`
      ]
        .filter(Boolean)
        .join(" and ")
    );
    return media;
  }, {} as Media);

  return conf;
};

export default function config(props?: ConfigProps): Config {
  const customConf = hasCustomConf(props);
  if (configs[0] === customConf) {
    return configs[1];
  }

  const conf = resolveConfig(props);

  configs[0] = customConf;
  configs[1] = conf;

  return conf;
}

function makeMedia(media: string) {
  return (...args: CSSObject[] | TemplateStringsArray) => css`
    @media ${media} {
      ${css.apply(null, [...args])}
    }
  `;
}
