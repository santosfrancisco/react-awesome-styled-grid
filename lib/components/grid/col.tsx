import styled, { css } from "styled-components";
import config, { DIMENSIONS } from "../../config";
import { GridColProps } from "@site/types";

export const Col = styled.div<GridColProps>`
  box-sizing: border-box;
  flex: 1 0 auto;
  max-width: 100%;
  display: flex;
  flex-direction: column;

  ${(props) =>
    !props.noGutter &&
    css`
      ${DIMENSIONS.map(
        (dimension) =>
          config(props).breakpoints[dimension] &&
          config(props).media[dimension]`
      padding-right: ${config(props).gutterWidth[dimension] / 2}rem;
      padding-left: ${config(props).gutterWidth[dimension] / 2}rem;
    `
      )}
    `}

  ${(props) => css`
    ${DIMENSIONS.map(
      (dimension) =>
        props[dimension] &&
        config(props).breakpoints[dimension] &&
        config(props).media[dimension]`
      flex: 1 1 ${(+props[dimension] / config(props).columns[dimension]) * 100}%;
      max-width: ${(+props[dimension] / config(props).columns[dimension]) * 100}%;
    `
    )}
  `}

  ${(props) =>
    props.offset &&
    css`
      ${DIMENSIONS.map(
        (dimension) =>
          config(props).breakpoints[dimension] &&
          config(props).media[dimension]`
    ${
      typeof props.offset === "object"
        ? props.offset[dimension] !== undefined &&
          `margin-left: ${props.offset[dimension] > 0 ? (props.offset[dimension] / config(props).columns[dimension]) * 100 : 0}%`
        : `margin-left: ${(props.offset / config(props).columns["xs"]) * 100}%`
    };
    `
      )}
    `}

  ${(props) =>
    props.order !== undefined &&
    css`
      ${DIMENSIONS.map(
        (dimension) =>
          config(props).breakpoints[dimension] &&
          config(props).media[dimension]`
    ${
      typeof props.order === "object"
        ? props.order[dimension] !== undefined &&
          `order: ${props.order[dimension]}`
        : `order: ${props.order}`
    };
    `
      )}
    `}

  ${(props) =>
    props.reverse &&
    css`
      ${Array.isArray(props.reverse)
        ? DIMENSIONS.map(
            (dimension) =>
              config(props).breakpoints[dimension] &&
              config(props).media[dimension]`
        flex-direction: ${Array.isArray(props.reverse) && props.reverse.indexOf(dimension) !== -1 ? "column-reverse" : "column"};
      `
          )
        : "flex-direction: column-reverse;"}
    `}

  ${(props) =>
    props.align &&
    css`
      ${typeof props.align === "object"
        ? DIMENSIONS.map(
            (dimensions) =>
              config(props).breakpoints[dimensions] &&
              config(props).media[
                dimensions
              ]`${props.align[dimensions] && `align-items: ${props.align[dimensions]}`}`
          )
        : `align-items: ${props.align};`}
    `}
  
  ${(props) =>
    props.justify &&
    css`
      ${typeof props.justify === "object"
        ? DIMENSIONS.map(
            (dimensions) =>
              config(props).breakpoints[dimensions] &&
              config(props).media[
                dimensions
              ]`${props.justify[dimensions] && `justify-content: ${props.justify[dimensions]}`}`
          )
        : `justify-content: ${props.justify};`}
    `}
  
  ${({ debug }) =>
    debug &&
    css`
      background-color: #5901ad40;
      outline: #fff solid 1px;
    `}
`;
