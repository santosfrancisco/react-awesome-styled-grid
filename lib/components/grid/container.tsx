import styled, { css } from "styled-components";
import isPropValid from "@emotion/is-prop-valid";
import config, { DIMENSIONS } from "../../config";
import { GridContainerProps } from "../../types";

export const Container = styled("div").withConfig({
  shouldForwardProp: (prop) => isPropValid(prop)
})<GridContainerProps>`
  margin-right: auto;
  margin-left: auto;
  max-width: 100%;
  width: 100%;
  box-sizing: border-box;

  ${(props) => css`
    ${DIMENSIONS.map(
      (d) =>
        config(props).container[d] &&
        config(props).media[d]`
      padding-left: ${config(props).paddingWidth[d]}rem;
      padding-right: ${config(props).paddingWidth[d]}rem;
    `
    )}
  `}

  ${(props) =>
    !props.fluid &&
    css`
      ${DIMENSIONS.map(
        (d) =>
          config(props).container[d] &&
          config(props).media[d]`
      ${
        typeof config(props).container[d] === "number"
          ? `max-width: ${config(props).container[d]}rem;`
          : `width: 100%;`
      }
    `
      )}
    `}

  ${({ debug }) =>
    debug &&
    css`
      background-color: #5901ad40;
      outline: #fff solid 1px;
    `}
`;
