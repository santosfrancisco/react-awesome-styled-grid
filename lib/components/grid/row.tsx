import styled, { css } from "styled-components";
import config, { DIMENSIONS } from "../../config";
import { GridRowProps } from "@site/types";

export const Row = styled.div<GridRowProps>`
  box-sizing: border-box;
  display: flex;
  flex: 1 1 auto;
  flex-wrap: wrap;

  ${(props) => css`
    ${DIMENSIONS.map(
      (dimension) =>
        config(props).container[dimension] &&
        config(props).media[dimension]`
      margin-left: -${config(props).gutterWidth[dimension] / 2}rem;
      margin-right: -${config(props).gutterWidth[dimension] / 2}rem;
    `
    )}
  `}

  ${(props) =>
    props.reverse &&
    css`
      ${Array.isArray(props.reverse)
        ? DIMENSIONS.map(
            (dimension) =>
              config(props).breakpoints[dimension] &&
              config(props).media[dimension]`
         ${
           Array.isArray(props.reverse) &&
           props.reverse.indexOf(dimension) !== -1
             ? `
             flex-direction: row-reverse;
             flex-wrap: wrap-reverse;
             `
             : `
                 flex-direction: row;
                 flex-wrap: wrap;
               `
         };
        
      `
          )
        : `
            flex-direction: row-reverse;
            flex-wrap: wrap-reverse;
          `}
    `}

  ${(props) =>
    props.align &&
    css`
      ${typeof props.align === "object"
        ? DIMENSIONS.map(
            (dimension) =>
              config(props).breakpoints[dimension] &&
              config(props).media[
                dimension
              ]`${props.align[dimension] && `align-items: ${props.align[dimension]}`}`
          )
        : `align-items: ${props.align};`}
    `}
  
  ${(props) =>
    props.justify &&
    css`
      ${typeof props.justify === "object"
        ? DIMENSIONS.map(
            (dimension) =>
              config(props).breakpoints[dimension] &&
              config(props).media[dimension]`${
                props.justify[dimension] &&
                `justify-content: ${props.justify[dimension]}`
              }`
          )
        : `justify-content: ${props.justify};`}
    `}

  ${({ debug }) =>
    debug &&
    css`
      background-color: #5901ad40;
      outline: #fff solid 1px;
    `}
`;
