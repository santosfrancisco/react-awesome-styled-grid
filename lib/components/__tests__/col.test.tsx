import { render } from "@testing-library/react";
import { Col } from "../grid/col";
import { GridBreakpoints } from "@site/lib/types";

describe("<Col />", () => {
  it("should render default style correctly", () => {
    const { asFragment } = render(<Col>children</Col>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should have no gutter", () => {
    const { asFragment } = render(<Col noGutter>children</Col>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should have the specified size", () => {
    const props = {
      xs: 1,
      sm: 2,
      md: 3,
      lg: 4,
      xl: 5
    };

    const { asFragment } = render(<Col {...props}>children</Col>);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should have a offset space for each media", () => {
    const props = {
      offset: {
        xs: 1,
        sm: 2,
        md: 0,
        lg: 4,
        xl: 5
      }
    };

    const { asFragment } = render(<Col {...props}>children</Col>);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should have a offset space for all media", () => {
    const props = {
      offset: 2
    };

    const { asFragment } = render(<Col {...props}>children</Col>);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should have a order space for each media", () => {
    const props = {
      order: {
        xs: 5,
        sm: 3,
        md: 2,
        lg: 6,
        xl: 8
      }
    };

    const { asFragment } = render(<Col {...props}>children</Col>);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should have a order space for all media", () => {
    const props = {
      order: 3
    };

    const { asFragment } = render(<Col {...props}>children</Col>);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should have a reverse direction for all medias", () => {
    const { asFragment } = render(<Col reverse>children</Col>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should have the css rule justify-content center", () => {
    const props = {
      justify: "center"
    };
    const { asFragment } = render(<Col {...props}>children</Col>);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should have the css rule justify-content center only in MD screen", () => {
    const props = {
      justify: {
        md: "center"
      }
    };

    const { asFragment } = render(<Col {...props}>children</Col>);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should have the css rule align-items center", () => {
    const props = {
      align: "center"
    };
    const { asFragment } = render(<Col {...props}>children</Col>);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should have the css rule align-items center only in MD screen", () => {
    const props = {
      align: {
        md: "center"
      }
    };

    const { asFragment } = render(<Col {...props}>children</Col>);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should have a reverse direction only for all media", () => {
    const props = {
      reverse: true
    };
    const { asFragment } = render(<Col {...props}>children</Col>);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should have a reverse direction only for MD media", () => {
    const props = {
      reverse: ["md"] as GridBreakpoints[]
    };
    const { asFragment } = render(<Col {...props}>children</Col>);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should have different style when it`s debug props is true", () => {
    const { asFragment } = render(<Col debug>children</Col>);

    expect(asFragment()).toMatchSnapshot();
  });
});
