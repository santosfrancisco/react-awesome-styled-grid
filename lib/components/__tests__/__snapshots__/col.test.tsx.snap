// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Col /> should have a offset space for all media 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  flex: 1 0 auto;
  max-width: 100%;
  display: flex;
  flex-direction: column;
}

@media only screen and (min-width: 1rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 48rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 90rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 120rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 1rem) {
  .c0 {
    margin-left: 50%;
  }
}

@media only screen and (min-width: 48rem) {
  .c0 {
    margin-left: 50%;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    margin-left: 50%;
  }
}

@media only screen and (min-width: 90rem) {
  .c0 {
    margin-left: 50%;
  }
}

@media only screen and (min-width: 120rem) {
  .c0 {
    margin-left: 50%;
  }
}

<div
    class="c0"
    offset="2"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`<Col /> should have a offset space for each media 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  flex: 1 0 auto;
  max-width: 100%;
  display: flex;
  flex-direction: column;
}

@media only screen and (min-width: 1rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 48rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 90rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 120rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 1rem) {
  .c0 {
    margin-left: 25%;
  }
}

@media only screen and (min-width: 48rem) {
  .c0 {
    margin-left: 25%;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    margin-left: 0%;
  }
}

@media only screen and (min-width: 90rem) {
  .c0 {
    margin-left: 33.33333333333333%;
  }
}

@media only screen and (min-width: 120rem) {
  .c0 {
    margin-left: 41.66666666666667%;
  }
}

<div
    class="c0"
    offset="[object Object]"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`<Col /> should have a order space for all media 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  flex: 1 0 auto;
  max-width: 100%;
  display: flex;
  flex-direction: column;
}

@media only screen and (min-width: 1rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 48rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 90rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 120rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 1rem) {
  .c0 {
    order: 3;
  }
}

@media only screen and (min-width: 48rem) {
  .c0 {
    order: 3;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    order: 3;
  }
}

@media only screen and (min-width: 90rem) {
  .c0 {
    order: 3;
  }
}

@media only screen and (min-width: 120rem) {
  .c0 {
    order: 3;
  }
}

<div
    class="c0"
    order="3"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`<Col /> should have a order space for each media 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  flex: 1 0 auto;
  max-width: 100%;
  display: flex;
  flex-direction: column;
}

@media only screen and (min-width: 1rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 48rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 90rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 120rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 1rem) {
  .c0 {
    order: 5;
  }
}

@media only screen and (min-width: 48rem) {
  .c0 {
    order: 3;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    order: 2;
  }
}

@media only screen and (min-width: 90rem) {
  .c0 {
    order: 6;
  }
}

@media only screen and (min-width: 120rem) {
  .c0 {
    order: 8;
  }
}

<div
    class="c0"
    order="[object Object]"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`<Col /> should have a reverse direction for all medias 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  flex: 1 0 auto;
  max-width: 100%;
  display: flex;
  flex-direction: column;
  flex-direction: column-reverse;
}

@media only screen and (min-width: 1rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 48rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 90rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 120rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

<div
    class="c0"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`<Col /> should have a reverse direction only for MD media 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  flex: 1 0 auto;
  max-width: 100%;
  display: flex;
  flex-direction: column;
}

@media only screen and (min-width: 1rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 48rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 90rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 120rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 1rem) {
  .c0 {
    flex-direction: column;
  }
}

@media only screen and (min-width: 48rem) {
  .c0 {
    flex-direction: column;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    flex-direction: column-reverse;
  }
}

@media only screen and (min-width: 90rem) {
  .c0 {
    flex-direction: column;
  }
}

@media only screen and (min-width: 120rem) {
  .c0 {
    flex-direction: column;
  }
}

<div
    class="c0"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`<Col /> should have a reverse direction only for all media 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  flex: 1 0 auto;
  max-width: 100%;
  display: flex;
  flex-direction: column;
  flex-direction: column-reverse;
}

@media only screen and (min-width: 1rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 48rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 90rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 120rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

<div
    class="c0"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`<Col /> should have different style when it\`s debug props is true 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  flex: 1 0 auto;
  max-width: 100%;
  display: flex;
  flex-direction: column;
  background-color: #5901ad40;
  outline: #fff solid 1px;
}

@media only screen and (min-width: 1rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 48rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 90rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 120rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

<div
    class="c0"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`<Col /> should have no gutter 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  flex: 1 0 auto;
  max-width: 100%;
  display: flex;
  flex-direction: column;
}

<div
    class="c0"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`<Col /> should have the css rule align-items center 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  flex: 1 0 auto;
  max-width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

@media only screen and (min-width: 1rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 48rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 90rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 120rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

<div
    class="c0"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`<Col /> should have the css rule align-items center only in MD screen 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  flex: 1 0 auto;
  max-width: 100%;
  display: flex;
  flex-direction: column;
}

@media only screen and (min-width: 1rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 48rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 90rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 120rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    align-items: center;
  }
}

<div
    class="c0"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`<Col /> should have the css rule justify-content center 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  flex: 1 0 auto;
  max-width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

@media only screen and (min-width: 1rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 48rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 90rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 120rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

<div
    class="c0"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`<Col /> should have the css rule justify-content center only in MD screen 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  flex: 1 0 auto;
  max-width: 100%;
  display: flex;
  flex-direction: column;
}

@media only screen and (min-width: 1rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 48rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 90rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 120rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    justify-content: center;
  }
}

<div
    class="c0"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`<Col /> should have the specified size 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  flex: 1 0 auto;
  max-width: 100%;
  display: flex;
  flex-direction: column;
}

@media only screen and (min-width: 1rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 48rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 90rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 120rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 1rem) {
  .c0 {
    flex: 1 1 25%;
    max-width: 25%;
  }
}

@media only screen and (min-width: 48rem) {
  .c0 {
    flex: 1 1 25%;
    max-width: 25%;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    flex: 1 1 37.5%;
    max-width: 37.5%;
  }
}

@media only screen and (min-width: 90rem) {
  .c0 {
    flex: 1 1 33.33333333333333%;
    max-width: 33.33333333333333%;
  }
}

@media only screen and (min-width: 120rem) {
  .c0 {
    flex: 1 1 41.66666666666667%;
    max-width: 41.66666666666667%;
  }
}

<div
    class="c0"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`<Col /> should render default style correctly 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  flex: 1 0 auto;
  max-width: 100%;
  display: flex;
  flex-direction: column;
}

@media only screen and (min-width: 1rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 48rem) {
  .c0 {
    padding-right: 0.5rem;
    padding-left: 0.5rem;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 90rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

@media only screen and (min-width: 120rem) {
  .c0 {
    padding-right: 0.75rem;
    padding-left: 0.75rem;
  }
}

<div
    class="c0"
  >
    children
  </div>
</DocumentFragment>
`;
